### DISCUSSION TITLE
Array validation function is getting a nuimber instead of an array

### DISCUSSION THREAD

**Discussion by ahmedHusseinF** [payloadcms/payload#1785] [Feature Requests & Ideas]
# Bug Report

When an array is nested in a tab, validation is broken while creating/editing.
The array value is passed as a number probably its size.

If this is a real issue, it might trigger a breaking change with its fix

## Steps to Reproduce

There is a [demo repo](https://github.com/ahmedHusseinF/payload-nested-array) where this is reproduced.
navigate to [http://localhost:3300](http://localhost:3300)

1. Have a top-level Tab structure
2. Have an Array as a field in a Tab
3. Have a dummy field inside the array
4. The array validation gets a value of either undefined/(potentially array size as number)

## Other Details

Payload v1.4.1
Firefox v108.0

**Comment by jmikrut** [comment]
Hey @ahmedHusseinF — this is indeed deliberate. The `array` field itself validates based on how many rows it has, although we could potentially update this in the future like you've mentioned.

In the meantime, the `validate` field does indeed accept `siblingData` as a prop, so you should be able to access the data for the entire array and still write a custom validation function that is based on the array's data in its entirety. 

Would that solve for what you're looking to do? I will bump this over to a feature request and we can keep the conversation going there!
